src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -113,6 +113,9 @@ void M(C c)
             var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8,
                 references: new[] { useMetadataImage ? lib.EmitToImageReference() : lib.ToMetadataReference() });
             comp.VerifyEmitDiagnostics(
+                // (6,45): error CS8852: Init-only property or indexer 'C.Property' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
+                //         _ = new Container() { contained = { Property = string.Empty, Property2 = string.Empty } };
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "Property").WithArguments("C.Property").WithLocation(6, 45),
                 // (6,45): error CS8400: Feature 'init-only setters' is not available in C# 8.0. Please use language version 9.0 or greater.
```

Comment:
>// (6,45): error CS8400: Feature 'init-only setters' is not available in C# 8.0. Please use language version 9.0 or greater. [](start = 16, length = 124)

I am not sure it is worth reporting this error here at all. No matter what version is used, an init-only property cannot-be assigned here. I'd say it is probably better to not make any changes in ```CheckPropertyValueKind```.